File: src/MyServiceBus.Http/HttpSendContext.cs
using System;
using System.Threading;
using MyServiceBus.Serialization;

namespace MyServiceBus;

public class HttpSendContext : SendContext
{
    public HttpSendContext(Type[] messageTypes, IMessageSerializer serializer, CancellationToken cancellationToken = default)
        : base(messageTypes, serializer, cancellationToken)
    {
    }

    public string ContentType { get; set; } = "application/json";
}

File: src/Java/myservicebus/src/main/java/com/myservicebus/HttpSendContext.java
package com.myservicebus;

import com.myservicebus.tasks.CancellationToken;

public class HttpSendContext extends SendContext {
    private String contentType = "application/json";

    public HttpSendContext(Object message, CancellationToken cancellationToken) {
        super(message, cancellationToken);
    }

    public String getContentType() {
        return contentType;
    }

    public void setContentType(String contentType) {
        this.contentType = contentType;
    }
}

